ImputationPipeline/Validation/ValidateScoring.wdl

```diff
@@ -0,0 +1,264 @@
+version 1.0
+
+import "../ScoringPart.wdl" as Scoring
+import "https://raw.githubusercontent.com/broadinstitute/palantir-workflows/main/ImputationPipeline/ScoringPart.wdl" as ScoringMain
+
+workflow ValidateScoring {
+	input {
+		File? validationArrays
+		File validationArraysMain
+		File validationWgs
+
+		String population_basename
+
+		File population_loadings
+		File population_meansd
+		File population_pcs
+		File pruning_sites_for_pca # and the sites used for PCA
+		File population_vcf
+
+		File weights
+		File sample_name_map
+		String branch
+
+		Int wgs_vcf_to_plink_mem = 8
+	}
+
+
+
+	call ExtractIDs as extractImputedIDs {
+		input:
+			vcf = select_first([validationArrays, validationArraysMain]),
+			output_basename = "imputed"
+	}
+
+	call ExtractIDs as extractImputedIDsMain {
+		input:
+			vcf = validationArraysMain,
+			output_basename = "imputed"
+	}
+
+
+	call SubsetWeights {
+		input:
+			sites = extractImputedIDs.ids,
+			weights = weights
+	}
+
+	call SubsetWeights as SubsetWeightsMain {
+		input:
+			sites = extractImputedIDsMain.ids,
+			weights = weights
+	}
+
+
+	call Scoring.ScoringImputedDataset as ScoreImputed {
+		input:
+			weights = weights,
+			imputed_array_vcf = select_first([validationArrays, validationArraysMain]),
```

Comment:
Why would validationArrays be optional (and fall back to the one in main) but validationArraysMain not?

Reply:
we need at least on set input array vcf to test on.  There are two basic testing setups I'm trying to cover here:
1) We want to test just the difference in the scoring pipeline between the new branch and main.  In this case, we want to run the scoring on the same vcf with both branches.  It doesn't really matter whether the shared input in this case is validationArrays or validationArraysMain, I just picked one.
2) We want to test the differences that changes to both the imputation and scoring pipelines together (or more likely, changes to imputation with unchanged scoring) have on prs scores.  In this case we need to have two different inputs for the results of imputation on the two different branches.